daml/splice-wallet/daml/Splice/Wallet/MintingDelegation.daml

```diff
@@ -0,0 +1,149 @@
+           dso
+           beneficiary
+           delegate
+           expiresAt = now `addRelTime` getMintingDelegationLifetime config
+           amuletMergeLimit = defaultAmuletMergeLimit
+
+    choice MintingDelegationProposal_Reject : MintingDelegationProposal_RejectResult
+      controller delegate
+      do pure MintingDelegationProposal_RejectResult with dummyUnitField = ()
+
+    choice MintingDelegationProposal_Withdraw : MintingDelegationProposal_WithdrawResult
+      controller beneficiary
+      do pure MintingDelegationProposal_WithdrawResult with dummyUnitField = ()
+
+defaultAmuletMergeLimit : Int
+defaultAmuletMergeLimit = 10
+
+template MintingDelegation
+  with
+    dso : Party
+    beneficiary : Party
+    delegate : Party
+    expiresAt : Time
+    amuletMergeLimit : Int
+      -- ^ The number of amulet contracts to keep after auto-merging; i.e.,
+      -- auto-merge contracts once there are strictly more than this number of contracts.
+  where
+    signatory beneficiary, delegate
+
+    nonconsuming choice MintingDelegation_RecordValidatorLivenessActivity : MintingDelegation_RecordValidatorLivenessActivityResult
+      with
+        validatorLicenseCid : ContractId ValidatorLicense
+        openRoundCid : ContractId OpenMiningRound
+      controller delegate
+      do assertWithinDeadline "expiresAt" expiresAt
+         _ <- fetchChecked (ForOwner with dso, owner = beneficiary) validatorLicenseCid
+         validatorLivenessResult <- exercise validatorLicenseCid (ValidatorLicense_RecordValidatorLivenessActivity openRoundCid)
+         pure MintingDelegation_RecordValidatorLivenessActivityResult with validatorLivenessResult
+
+    nonconsuming choice MintingDelegation_Mint : MintingDelegation_MintResult
+      with
+        inputs : [TransferInput]
+        context : PaymentTransferContext
+      controller delegate
+      do assertWithinDeadline "expiresAt" expiresAt
+         require "Minting is unfeatured" (context.context.featuredAppRight == None)
+         transferResult <- exercise context.amuletRules AmuletRules_Transfer with
+           transfer = Transfer with
+             sender = beneficiary
+             provider = delegate
+             inputs
+             outputs = []
+             beneficiaries = None
+           context = context.context
+           expectedDso = Some dso
+         pure MintingDelegation_MintResult with transferResult
+
+    choice MintingDelegation_Reject : MintingDelegation_RejectResult
+      controller delegate
+      do pure MintingDelegation_RejectResult with dummyUnitField = ()
+
+    choice MintingDelegation_Withdraw : MintingDelegation_WithdrawResult
+      controller beneficiary
+      do pure MintingDelegation_WithdrawResult with dummyUnitField = ()
+
+    -- We do not charge a fee for MintingDelegation but we make them expire
+    -- and force paying traffic to renew them to protect against attacks.
```

Comment:
We can remove this choice now that the `DSO` is no longer involved in the delegation.